www/apps/resources/app/recipes/marketplace/examples/vendors/page.mdx

```diff
@@ -0,0 +1,1344 @@
+import { Github, PlaySolid } from "@medusajs/icons"
+
+export const metadata = {
+  title: `Marketplace Recipe: Vendors Example`,
+}
+
+# {metadata.title}
+
+This document provides an example of implementing the marketplace recipe.
+
+<Note>
+
+You can implement the marketplace as you see fit for your use case. This is only an example of one way to implement it.
+
+</Note>
+
+## Features
+
+By following this example, you’ll have a marketplace with the following features:
+
+1. Multiple vendors, each having vendor admins.
+2. Vendor admins manage the vendor’s products and orders.
+3. On order creation, the order is split into multiple orders for each vendor.
+4. All other commerce features that Medusa provides.
+
+<CardList items={[
+  {
+    href: "https://github.com/medusajs/examples/tree/main/marketplace",
+    title: "Marketplace Example Repository",
+    text: "Find the full code for this recipe example in this repository.",
+    startIcon: <Github />,
+    showLinkIcon: false
+  },
+  {
+    href: "https://res.cloudinary.com/dza7lstvk/raw/upload/v1720603521/OpenApi/Marketplace_OpenApi_n458oh.yml",
+    title: "OpenApi Specs for Postman",
+    text: "Imported this OpenApi Specs file into tools like Postman.",
+    startIcon: <PlaySolid />,
+    showLinkIcon: false
+  },
+]} />
+
+---
+
+<Note type="check">
+
+- [A new Medusa application installed.](!docs!#get-started)
+
+</Note>
+
+## Step 1: Create Marketplace Module
+
+The first step is to create a marketplace module that holds the data models for a vendor and an admin. Create the directory `src/modules/marketplace`.
+
+Then, create the file `src/modules/marketplace/models/vendor.ts` with the following content:
+
+```ts title="src/modules/marketplace/models/vendor.ts"
+import { model } from "@medusajs/utils"
+import VendorAdmin from "./vendor-admin"
+
+const Vendor = model.define("vendor", {
+  id: model.id().primaryKey(),
+  handle: model.text().unique(),
+  name: model.text(),
+  logo: model.text().nullable(),
+  admins: model.hasMany(() => VendorAdmin)
+})
+
+export default Vendor
+```
+
+This creates a `Vendor` data model, which represents a business that sells its products in the marketplace.
+
+Notice that the `Vendor` has many admins whose data model you’ll create next.
+
+Create the file `src/modules/marketplace/models/vendor-admin.ts` with the following content:
+
+```ts title="src/modules/marketplace/models/vendor-admin.ts"
+import { model } from "@medusajs/utils"
+import Vendor from "./vendor"
+
+const VendorAdmin = model.define("vendor_admin", {
+  id: model.id().primaryKey(),
+  first_name: model.text().nullable(),
+  last_name: model.text().nullable(),
+  email: model.text().unique(),
+  vendor: model.belongsTo(() => Vendor, {
+    mappedBy: "admins"
+  })
+})
+
+export default VendorAdmin
+```
+
+This creates a `VendorAdmin` data model, which represents an admin of a vendor.
+
+Then, add a migration file at `src/modules/marketplace/migrations/Migration20240708151444.ts`:
```

Comment:
question: Should we not be teaching them about generating migrations instead of providing them with a copy/paste?

It's more error-prone, but if the generated migrations leads to issues, then there is something we need to fix elsewhere.

Reply:
Ah I see. I just thought it would be nice to get the workflow to generate migrations under their skin. The next step for them after this Recip would probably be to create more models or modules, and then they wouldn't have a migration to copy over.

Anyway, I'll let it be up to you 👍

Reply:
Makes sense when you put it that way! I changed the section to generate the migration